fx-src/FirefoxStorage.js

```diff
@@ -197,6 +202,39 @@ export default class FirefoxAdapter extends BaseAdapter {
     });
   }
 
+  loadDump(records) {
+    let connection = this._connection;
+    let collection_name = this.collection;
+    return Task.spawn(function* () {
+      yield connection.executeTransaction(function* doImport() {
+        for (let record of records) {
+          const params = {
+            collection_name: collection_name,
+            record_id: record.id,
+            record: JSON.stringify(record)
+          };
+          yield connection.execute(statements.importData, params);
```

Comment:
I think you probably want a two-phase operation here instead. Either

```
SELECT last_modified, record_id FROM records WHERE collection_name = :collection_name AND record_id IN (?, ?, ?, …)
```

then merge in memory and do `INSERT OR REPLACE` with just the records that are newer or not present, or

```
UPDATE … WHERE last_modified < :last_modified
```

for each incoming record, checking the number of modified rows; if none modified, then

```
INSERT OR IGNORE …
```

to insert missing records.